tests/test_docker_validator.py

```diff
@@ -62,28 +64,35 @@ def setUp(self):
 
         self.validator_from_excel = DockerValidator(
             mapping_file=self.mapping_file,
-            submission_dir=self.submission_dir,
+            submission_dir=self.test_run_dir2,
             project_title=self.project_title,
-            metadata_xlsx=self.metadata_xlsx,
-            container_name='eva-sub-cli-test'
+            metadata_xlsx=self.metadata_xlsx
         )
 
     def tearDown(self):
-        if os.path.exists(self.test_run_dir):
-            shutil.rmtree(self.test_run_dir)
+        for d in [self.test_run_dir1, self.test_run_dir2]:
+            if os.path.exists(d):
+                shutil.rmtree(d)
         self.validator.stop_running_container()
         self.validator_from_excel.stop_running_container()
 
+    def assert_same_dict_and_unordered_list(self, o1, o2):
+        if isinstance(o1, dict) and isinstance(o2, dict):
+            self.assertEqual(set(o1), set(o2))
+            [self.assert_same_dict_and_unordered_list(o1.get(k), o2.get(k)) for k in o1]
+        elif isinstance(o1, list) and isinstance(o2, list):
+            self.assertEqual(set(o1), set(o2))
+        else:
+            self.assertEqual(o1, o2)
+
     def assert_sample_checker(self, sample_checker_file, expected_checker):
         self.assertTrue(os.path.isfile(sample_checker_file))
         with open(sample_checker_file) as open_yaml:
-            assert yaml.safe_load(open_yaml) == expected_checker
-
-    def test_validate(self):
-        # run validation in docker
-        self.validator.validate()
+            self.assert_same_dict_and_unordered_list(yaml.safe_load(open_yaml), expected_checker)
 
-        vcf_format_dir = os.path.join(self.validator.output_dir, 'vcf_format')
+    def assess_post_validation(self, validator, expected_sample_checker, expected_metadata_files_json,
```

Comment:
```suggestion
    def assert_validation_results(self, validator, expected_sample_checker, expected_metadata_files_json,
```
I don't know if this is a common thing or just me, but I kind of like these helper methods to have "assert" in the name - this way when I'm reading the tests, I can easily pick out where the assertions are happening. Totally optional though.